joblib/memory.py

```diff
@@ -889,6 +889,23 @@ class Memory(Logger):
             **Note:** You need to call :meth:`joblib.Memory.reduce_size` to
             actually reduce the cache size to be less than ``bytes_limit``.
 
+        files_limit: int, optional
+            Number of files to limit the cache to.  By default, the number of
+            files in the cache is unlimited. When reducing the size of the
+            cache, ``joblib`` keeps the most recently accessed items first.
+
+            **Note:** You need to call :meth:`joblib.Memory.reduce_size` to
+            actually reduce the number of files to be less than
+            ``files_limit``.
+
+        age_limit: naive datetime, optional
+            Maximum age of entries to limit the cache to.  When reducing the
```

Comment:
since it is a `datetime` (and not `timedelta`), not sure if `age_limit` and "maximum age" (would be thought to be e.g. "a day", "a year") describes it well. Also because it is based on `last_access` datetime and not some `created_datetime`, I wonder if `age` is appropriate... If we were to keep it as datetime thinking of names:

- `last_access_limit` (or just `access_limit`) to parallel reliance of `last_access`
- better ideas?

and then tune up the docstring here to describe that it should be a date, not "age".  But may be use of  `timedelta` would be more convenient since it would be nice to be able to just say `timedelta(days=10)` instead of doing some `now() - timedelta(days=10)`. Then name could be the same `last_access_limit`.  

So altogether, may be best would be actually to just support both `datetime` and `timedelta` and do the `now() - last_access_limit` if timedelta provided?

Reply:
I would like to hear from a joblib maintainer on this.

Reply:
Dear @ogrisel @GaelVaroquaux et al - please guide us to get to the API you would like to see here.